server/datastore/mysql/scim_test.go

```diff
@@ -1135,10 +1146,226 @@ func testScimUserCreateValidation(t *testing.T, ds *Datastore) {
 		FamilyName: ptr.String("Name"),
 		Active:     ptr.Bool(true),
 	}
-	_, err = ds.CreateScimUser(context.Background(), &validUser)
+	_, err = ds.CreateScimUser(t.Context(), &validUser)
 	assert.NoError(t, err)
 }
 
+func testScimUserByHostID(t *testing.T, ds *Datastore) {
+	// Create a test SCIM user with emails
+	user1 := fleet.ScimUser{
+		UserName:   "host-test-user1",
+		ExternalID: ptr.String("ext-host-123"),
+		GivenName:  ptr.String("Host"),
+		FamilyName: ptr.String("User"),
+		Active:     ptr.Bool(true),
+		Emails: []fleet.ScimUserEmail{
+			{
+				Email:   "host.user@example.com",
+				Primary: ptr.Bool(true),
+				Type:    ptr.String("work"),
+			},
+		},
+	}
+
+	var err error
+	user1.ID, err = ds.CreateScimUser(t.Context(), &user1)
+	require.Nil(t, err)
+
+	// Create a group and associate it with the user
+	group := fleet.ScimGroup{
+		DisplayName: "Host Test Group",
+		ExternalID:  ptr.String("ext-host-group"),
+		ScimUsers:   []uint{user1.ID},
+	}
+	group.ID, err = ds.CreateScimGroup(t.Context(), &group)
+	require.Nil(t, err)
+
+	// Create a second test SCIM user without emails and without groups
+	user2 := fleet.ScimUser{
+		UserName:   "host-test-user2",
+		ExternalID: ptr.String("ext-host-456"),
+		GivenName:  ptr.String("No"),
+		FamilyName: ptr.String("Emails"),
+		Active:     ptr.Bool(true),
+		Emails:     []fleet.ScimUserEmail{},
+	}
+	user2.ID, err = ds.CreateScimUser(t.Context(), &user2)
+	require.Nil(t, err)
+
+	// Create test hosts
+	hostID1 := uint(1000001) // Use a dummy host ID for testing
+	hostID2 := uint(1000002) // Use a different dummy host ID for testing
```

Comment:
This is generally something that comes back to haunt us later, using "non-backed IDs" that don't represent real things in the DB (e.g. if with mysql 9 or something it becomes manageable to add FKs to hosts, or just some behavior change that makes a check that host exists). Not saying to change it here, but in general we should avoid this and create real hosts.